src/app/OperationalSessionSetup.cpp

```diff
@@ -553,27 +576,38 @@ CHIP_ERROR OperationalSessionSetup::ScheduleSessionSetupReattempt(System::Clock:
     return err;
 }
 
-void OperationalSessionSetup::TrySetupAgain(System::Layer * systemLayer, void * state)
+void OperationalSessionSetup::CancelSessionSetupReattempt()
 {
-    auto * self = static_cast<OperationalSessionSetup *>(state);
-
-    CHIP_ERROR err = CHIP_NO_ERROR;
+    // If we can't get a system layer, there is no way for us to cancel things
+    // at this point, but hopefully that's because everything is torn down
+    // anyway and hence the timer will not fire.
+    auto * sessionManager = mInitParams.exchangeMgr->GetSessionManager();
+    if (!sessionManager)
+    {
+        return;
+    }
 
-    if (self->mState != State::NeedsAddress)
+    auto * systemLayer = sessionManager->SystemLayer();
+    if (!systemLayer)
     {
-        err = CHIP_ERROR_INCORRECT_STATE;
+        return;
     }
-    else
+
+    systemLayer->CancelTimer(TrySetupAgain, this);
+}
+
+void OperationalSessionSetup::TrySetupAgain(System::Layer * systemLayer, void * state)
+{
+    auto * self = static_cast<OperationalSessionSetup *>(state);
+
+    self->MoveToState(State::ResolvingAddress);
+    CHIP_ERROR err = self->LookupPeerAddress();
+    if (err == CHIP_NO_ERROR)
     {
-        self->MoveToState(State::ResolvingAddress);
-        err = self->LookupPeerAddress();
-        if (err == CHIP_NO_ERROR)
-        {
-            return;
-        }
+        return;
```

Comment:
nit: `ReturnOnFailure(self->LookupPeerAddress());` ?

Reply:
No, we want to return if `LookupPeerAddress` _succeeds_.  On failure we want to execute other logic.  Leaving this as-is.